File: Sort_of_Sorting/Sort_of_Sorting.py
num = -1
while num != 0:
    num = int(input())
    if num == 0:
        break
    names = []
    for _ in range(num):
        name = input()
        names.append(name)
    names.sort(key=lambda x: x[:2])
    for name in names:
        print(name)
    print()

File: Sort_of_Sorting/Sort_of_Sorting.cpp
#include <algorithm>
#include <bits/stdc++.h>
#include <iostream>
#include <vector>
using namespace std;

bool twoCharacter(const string &a, const string &b)
{

    return a.substr(0, 2) < b.substr(0, 2);
}

int main()
{

    while (true)
    {
        int studentName;
        cin >> studentName;
        if (studentName == 0)
        {
            break;
        }

        vector<string> names;
        for (int i = 0; i < studentName; i++)
        {
            string name;
            cin >> name;
            names.push_back(name);
        }
        stable_sort(names.begin(), names.end(), twoCharacter);

        for (auto it = names.begin(); it != names.end(); it++)
        {
            cout << *it << "\n";
        }
        cout << "\n";
    }
    return 0;
}